lab_2_tokenize_by_bpe/main.py

```diff
@@ -58,7 +102,24 @@ def train(
     :param num_merges: required number of new tokens
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
-
+    if not isinstance(word_frequencies, dict) or not isinstance(num_merges, int):
+        return None
+    while num_merges > 0:
+        num_of_pairs = count_tokens_pairs(word_frequencies)
+        if num_of_pairs is None:
+            return None
+        if num_merges > len(num_of_pairs):
+            num_merges = len(num_of_pairs)
+        len_most_freq = max(num_of_pairs.values())
+        most_freq_pairs = [pair for pair, frequency in num_of_pairs.items() if frequency == len_most_freq]
+        max_length = max(len(str(pair)) for pair in most_freq_pairs)
+        longest_pairs = [pair for pair in most_freq_pairs if len(str(pair)) == max_length]
+        more_suitable_pairs = sorted(longest_pairs)
```

Comment:
Consider just sorting instead of those lists of most frequent pairs, longest pairs etc...

Reply:
Done. But I'm not sure that I'm right because now it passes one test less